### DISCUSSION TITLE
Gatsby SSR pages doesnt work with AWS Amplify Deployment

### DISCUSSION THREAD

**Discussion by shivaping** [gatsbyjs/gatsby#34266] [Help]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

Next JS SSR is supported on Aws Amplify but Gatsby SSR is not supported right now.

### Reproduction Link

https://www.marlin-staging-john-rogers.guidesly.com/ssr-page

### Steps to Reproduce

1.Followed article on https://www.gatsbyjs.com/docs/reference/config-files/gatsby-ssr/
2. created a page ssr-page under pages folder 
3. Deployed the site to https://www.marlin-staging-john-rogers.guidesly.com
4. tried browsing https://www.marlin-staging-john-rogers.guidesly.com/ssr-page. The page is not working 
...

### Expected Result

https://www.marlin-staging-john-rogers.guidesly.com/ssr-page should stay on the page and render the dynamic content

### Actual Result

https://www.marlin-staging-john-rogers.guidesly.com/ssr-page is redirected to home page.

### Environment

```shell
AWS Cloud Amplify

Node : 16.x
npm : 8.x
```

**Comment by LekoArts** [comment]
Hi!

Please reach out to AWS Amplify as they'll need to add support for SSR & DSG in Gatsby. In the meantime you can use Gatsby Cloud to have it fully functional working.

**Reply by juanbiberretta** [reply]
@LekoArts I was wondering something along these lines as well - we're self-hosting our static site assets on Google Cloud Buckets and after scouring the docs I still don't know how one gets a Gatsby "server" running for the new SSR feature. It's all well and good if we're just serving an `index.html` file, but how do we transition that to a server? Do we have to do `gatsby build` and then `gatsby serve` on running Node.js server? 

I understand Gatsby Cloud does this, but it's not clear _how_, nor can I find any docs on how you would run your own server, just how to deploy to all the different providers.

I guess my question is: what does it take, from a very bare-bones hosting perspective, to enable server-side data fetching and rendering? Does AWS Amplify need go from hosting static assets to _running_ a Node.js server? Seems like quite the ask 🤔 

Thank you!

**Reply by shivaping** [reply]
> Hi!
> 
> Please reach out to AWS Amplify as they'll need to add support for SSR & DSG in Gatsby. In the meantime you can use Gatsby Cloud to have it fully functional working.

Gatsby cloud is very expensive and we would like to stick with AWS or GCP. What's the alternative if we would proceed with GCP ?

**Reply by pragmaticpat** [reply]
Hi @shivaping 

Gatsby doesn’t officially support running SSR or DSG anywhere other than Gatsby Cloud at this time. However, there are community implementers that have made these features available on other platforms, which could be referenced in order to implement these features in any hosting environment you choose.

**☝️Note**: Since Gatsby doesn’t officially support these features outside of Gatsby Cloud, we cannot guarantee their performance or stability.

All that said, here are some useful references at this time:
- [gatsby-plugin-fastify](https://www.npmjs.com/package/gatsby-plugin-fastify) - this plugin can be deployed anywhere you choose to run a Fastify web server
- [netlify-plugin-gatsby](https://github.com/netlify/netlify-plugin-gatsby#readme) - only useful if you’re hosting with Netlify

**Reply by Patrickramosg** [reply]
Hi @pragmaticpat Gatsby 4 still doesn't support SSR using AWS Amplify, right? Do we have any predictions?

Thanks!
